packages/blob/src/index.ts

```diff
@@ -215,8 +211,60 @@ export async function list(
   };
 }
 
-function getApiUrl(): string {
-  return process.env.VERCEL_BLOB_API_URL || 'https://blob.vercel-storage.com';
+export interface GenerateClientTokenOptions extends BlobCommandOptions {
+  pathname: string;
+  onUploadCompletedUrl: string;
+  onUploadCompletedCallbackUrlArgs?: Record<string, string>;
+}
+
+export function generateClientTokenFromReadWriteToken({
+  pathname,
+  onUploadCompletedUrl,
+  token,
+  onUploadCompletedCallbackUrlArgs,
+}: GenerateClientTokenOptions): string {
+  const timestamp = new Date();
+  timestamp.setSeconds(timestamp.getSeconds() + 30);
+  const blobToken = getToken({ token });
+
+  const payload = Buffer.from(
+    JSON.stringify({
+      pathname,
+      validUntil: timestamp.getTime(),
+      onUploadCompletedUrl,
+      onUploadCompletedCallbackUrlArgs,
+    }),
+  ).toString('base64');
+
+  const [, , , storeId = null] = blobToken.split('_');
+
+  if (!storeId) {
+    throw new Error('Invalid BLOB_READ_WRITE_TOKEN');
+  }
+
+  const securedKey = crypto
+    .createHmac('sha256', blobToken)
+    .update(payload)
+    .digest('hex');
+
+  return `vercel_blob_client_${storeId}_${Buffer.from(
+    `${securedKey}.${payload}`,
+  ).toString('base64')}`;
+}
+
+type DecodedClientTokenPayload = Omit<GenerateClientTokenOptions, 'token'> & {
+  validUntil: number;
+};
+
+export function decodeClientToken(
```

Comment:
This gets exported at the package level, it should not be exported I believe?
If we need this for tests, we can have it in a different file instead and use it in tests only.

Also you could name it "getPayloadFromClientToken" since it doesn't really decode it but give you the raw payload part.

Reply:
I would export it and use on the vercel api, but I think it would be a useful helper for the user

Reply:
> but I think it would be a useful helper for the user
Curious: Any cases you can think of this would be useful for users?

Reply:
if you think about the jwt, the client can always access the payload and decode if needed